ReactiveCocoa/Swift/Signal.swift

```diff
@@ -1023,10 +1023,10 @@ extension SignalType {
 
 private struct ZipState<Left, Right> {
 	var values: (left: [Left], right: [Right]) = ([], [])
-	var completed: (left: Bool, right: Bool) = (false, false)
+	var isCompleted: (left: Bool, right: Bool) = (false, false)
 
 	var isFinished: Bool {
-		return (completed.left && values.left.isEmpty) || (completed.right && values.right.isEmpty)
+		return (isCompleted.left && values.left.isEmpty) || (isCompleted.right && values.right.isEmpty) || (isCompleted.left && isCompleted.right && (values.left.isEmpty || values.right.isEmpty))
```

Comment:
@mdiep Hmm. I was able to proof that this expression is the equivalent of the previous one. I would revert this.